src/main/java/com/mapzen/valhalla/Router.kt

```diff
@@ -0,0 +1,141 @@
+package com.mapzen.valhalla
+
+import com.google.common.base.Charsets
+import com.google.common.io.CharStreams
+import com.squareup.okhttp.OkHttpClient
+
+import org.json.JSONException
+
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.io.UnsupportedEncodingException
+import java.net.HttpURLConnection
+import java.net.MalformedURLException
+import java.net.URL
+import java.net.URLEncoder
+import java.util.ArrayList
+
+public class Router : Runnable {
+    private val DEFAULT_URL = "http://valhalla.api.dev.mapzen.com/route"
+    private val ROUTE_PARAMS = "?json={\"locations\":" + "[{\"lat\":%1f,\"lon\":%2f},{\"lat\":%3f,\"lon\":%4f}]," + "\"costing\":\"%s\",\"output\":\"json\"}"
+    private var endpoint = DEFAULT_URL
+    private val client = OkHttpClient()
+    private var type = Type.DRIVING
+    private val locations = ArrayList<DoubleArray>()
+    private var callback: Callback? = null
+    public enum class Type(private val type: String) {
+        WALKING : Type("pedestrian")
+        BIKING : Type("bicycle")
+        DRIVING : Type("auto")
+
+        override fun toString(): String {
+            return type
+        }
+    }
+
+    public fun getRouter(): Router {
```

Comment:
We should remove this method since its not possible to have static factory methods in Kotlin